src/client/ClientManager.js

```diff
@@ -32,26 +33,33 @@ class ClientManager {
   /**
    * Connects the client to the WebSocket.
    * @param {string} token The authorization token
-   * @param {Function} resolve Function to run when connection is successful
-   * @param {Function} reject Function to run when connection fails
+   * @param {Function} resolve Function to run when all shards are connected
+   * @param {Function} reject Function to run when a shard fails
    */
   connectToWebSocket(token, resolve, reject) {
+    if (this.client.options.internalSharding && this.client.shard) {
+      reject(new Error('SHARDING_CANNOT_USE_BOTH_METHODS'));
+      return;
+    }
     this.client.emit(Events.DEBUG, `Authenticated using token ${token}`);
     this.client.token = token;
-    const timeout = this.client.setTimeout(() => reject(new Error('WS_CONNECTION_TIMEOUT')), 1000 * 300);
-    this.client.api.gateway.get().then(async res => {
+    let timeout;
+    if (!this.client.options.internalSharding) {
+      timeout = this.client.setTimeout(() => reject(new Error('WS_CONNECTION_TIMEOUT')), 1000 * 300);
+    }
+    let endpoint = this.client.api.gateway;
+    if (this.client.options.internalSharding) endpoint = endpoint.bot;
+    endpoint.get({ forceBot: this.client.options.internalSharding }).then(async res => {
       if (this.client.options.presence != null) { // eslint-disable-line eqeqeq
         this.client.options.ws.presence = await this.client.presences._parse(this.client.options.presence);
       }
-      const gateway = `${res.url}/`;
+      if (res.shards) {
+        this.client.emit(Events.DEBUG, `Using recommended shard count: ${res.shards}`);
+        this.client.options.shardCount = res.shards;
+      }
+      const gateway = `${res.url}/?v${DefaultOptions.ws.version}&encoding=${WebSocket.encoding}`;
```

Comment:
The version is added in `WebSocketConnection#connect`, and as such should be removed. Same with encoding, as it is added in `WebSocket#create`.